### DISCUSSION TITLE
Request: 5x5 with BLTouch for a 410x410x450 Bed

### DISCUSSION THREAD

**Discussion by WheezerX** [Jyers/Marlin#1129] [General Discussion]
I recently got the Ender Extender 400 XL V2 kit and it's been successfully set up but my last piece of the puzzle is the firmware.

I'm unfortunately code illiterate and I have vague knowledge of it but not enough to be functional. Is there anyway I could get a BLTouch enabled, 410x410x450 print size firmware with the 5x5 feature?

**Comment by adam3654** [comment]
https://www.patreon.com/jyers

**Reply by WheezerX** [reply]
So if I support, he offers compiling services?

**Reply by smooreace** [reply]
The compile service is there... if you spent the time to LOOK at the info that was given to you you would see that. But you will still have to build the config files yourself

**Comment by Cal-LF** [comment]
Learn to compile your own FW. Coding is not needed to edit a text file using VS Code.

**Reply by WheezerX** [reply]
You know, you can just not post if you can't help.

**Reply by Reggi3** [reply]
It was an attempt to help, it didn't give you much information but it was definitely an attempt to help and encourage you to learn to compile your own firmware.  There are lots of tutorials for compiling marlin online, which is definitely worth the effort searching for, if you can manage it, not everyone can so there are alternatives such as becoming a Jyers patreon.  The main takeaway that I took from your original post and the previous reply were that you are 'code illiterate', pointing out that it doesn't require you to actually code anything is quite helpful as it means you'll only have to look at 2 files.  

The extra piece of information that you need is to know that to get JyersUI firmware (Or any marlin firmware for that matter) to work for you, you need to edit some #defines in the configs, these defines enable or disable certain features, they also set upper/lower limits and values that are specific to your machine setup.

The files you need to edit are in the Marlin/Configuration Files/ folder, I think E3V2 Templates, you can use either of the BLTouch 3x3 based configs and just search for GRID_MAX_POINTS_X.  Change the value from 3 to 5 and your grid issues are fixed.  All you need to do at this point is add in your bed dimension changes and you should be where you want to be or at least very close.

HTH.

**Reply by smooreace** [reply]
There is an immense thick line between help and hand-holding. The help offered was clear, concise, and 100% on point. Stop EXPECTING people to do everything for you and learn to invest so effort into yourself. It will eventually go a long way!! At least once you get past your current mentality of thinking the world owes you everything!

**Comment by WheezerX** [comment]
I'm kinda glad the neckbeards in Jyers community are so hostile and gatekeeping, I ended up going to Reddit and was immediately shown a superior firmware that let me edit the settings I needed in the actual LCD menu. Won't be coming back :)

**Reply by AktasC** [reply]
I feel like I'm talking for the whole community when I say I'm sincerely sorry for some of the answers you've been given by some individuals over here.    
If needed, I would gladly help you in your future issues.   
Jyers also replied, please read [his answer](https://github.com/Jyers/Marlin/discussions/1129#discussioncomment-981048) if you didn't already  

Regards

**Comment by smooreace** [comment]
Lol... And in an instant, the world is a better place! When he DOES come
back at least we won't see any more BS from him!

**Comment by Jyers** [comment]
Guys... This is horrible. This is not the community I know and certainly not what I would want.

@adam3654 Thanks for pointing him to a resource, but a little explanation would have been good about the auto-compile and why it would help his situation.

@smooreace There's really no need to be hostile about it. Even if there are some simple questions just being dismissive isn't really going to help anything. In fact, more often than not it's just going to create arguments which is the last thing I want.

@Cal-LF A valid point but a little extra direction would have been good.

It was a simple enough question, trying to answer it quickly or dismissively is just going to cause more confusion and make the discussion take longer than it needs to.
For this it would be as simple as letting them know they can get these features by compiling their own and maybe pass them a link to the wiki pages on compiling or on the video guides explaining how to do so.

@WheezerX I'm sorry that you received this kind of response, I can assure you the community here is generally quite accepting and helpful. It seems some of us have just gotten a tad complacent from the large influx of new users and questions. To fully answer your question, you can compile your own to change such settings. Info on the process can be found on the wiki [here (docs)](https://github.com/Jyers/Marlin/wiki/Configuring-&-Flashing) or [here (videos)](https://github.com/Jyers/Marlin/wiki/Video-Guides). Obviously you're welcome to use mriscoc's firmware if you would like, but if you ask me it's a bit of a money grab. He's mostly just been implementing the features I've already added, and he's still using the outdated DWIN code. I'm also hoping to eventually be able to contribute my changes to main Marlin repository, something he'll never be able to do with the way he's overriding configs. Also if you would like to use the autocompile service on my patreon, it has a simplified config option where you can use a much easier config that only has the options you'll really need, making it much easier to customize your firmware. As well as the added bonus of having firmware compiled for you every hour with the latest features. But obviously this is totally up to you, as it does require becoming a patron.

No matter what you end up doing, I just don't want you to look back at this community with a sour taste in your mouth. We are an active and driven community with the goal of making the best possible firmware for the Ender 3 V2. That is and will always be my goal for this group.

**Reply by Tuncay-Ayhan** [reply]
Great to hear you speak up. Talking through text can be interpreted several ways, but some people just love the discussions. They seem to live for it. Thanks for stepping in @Jyers. I couldn't agree more with you.

I think Github should be a place where you should be able to learn stuff at your own pace. Not someone trying to lecture you and try to be make you ask more intelligent questions. You should be able to ask anything in any way you want. 

No is also an answer. Just try not to be a d*ck about it. 

Not everyone is you. Not everyone has the same experience you have/had. Some people are just beginning and yes, you learn everyday. It doesn't matter what your age is.

If you don't feel like, or don't know the answer and all you can do is just say stuff to hurt someone's feelings, because they should better or search better, just don't answer. 

There is a saying:
**Give them a fish and they will feed themself 1 day. Teach them how to fish and they can feed themself for life.**

Buttt.. That isn't the case here, some fish evolve at a fast pace and some people don't even know what a fish is.

**Comment by smooreace** [comment]
Everything is "hostile" when the wrong someone doesn't get exactly what
they want. Bowing down to those that have an over-exaggerated imagination
is never the right solution. There is never anything wrong with asking
questions. But it's always wrong to that kind of person to give a clear
concise answer. A capable intelligent person takes that info and runs with
it to FIND the rest of the answer rather than DEMAND that the world is
against them for not spoon-feeding every action needed. A perfect example
comes from looking at the majority of today's younger "adults". Whats
needed is a virtual back-hand!
